File: app/src/main/java/com/jhughes/todoapp/injection/module/activity/SplashActivityModule.java
package com.jhughes.todoapp.injection.module.activity;

import android.app.Activity;
import android.app.Application;

import com.jhughes.todoapp.data.util.AppExecutors;
import com.jhughes.todoapp.injection.component.activity.SplashActivitySubComponent;
import com.jhughes.todoapp.ui.SplashActivity;
import com.jhughes.todoapp.ui.viewModel.factory.SplashViewModelFactory;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

@Module(subcomponents = SplashActivitySubComponent.class)
public abstract class SplashActivityModule {
    @Binds
    @IntoMap
    @ActivityKey(SplashActivity.class)
    public abstract AndroidInjector.Factory<? extends Activity>
    bindMainActivityInjectorFactory(SplashActivitySubComponent.Builder builder);

    @Provides
    public static SplashViewModelFactory provideSplashViewModelFactory(Application application, AppExecutors appExecutors) {
        return new SplashViewModelFactory(application, appExecutors);
    }
}

File: app/src/main/java/com/jhughes/todoapp/injection/module/activity/SplashActivityModule.kt
package com.jhughes.todoapp.injection.module.activity

import android.app.Application
import com.jhughes.todoapp.data.util.AppExecutors
import com.jhughes.todoapp.injection.scope.ActivityScope
import com.jhughes.todoapp.ui.viewModel.factory.SplashViewModelFactory
import dagger.Module
import dagger.Provides

//module doesnt contain any abstract android injectors, doesnt need to be abstract

@Module(includes = [BaseActivityModule::class])
class SplashActivityModule {

    @Provides
    @ActivityScope
    fun provideSplashViewModelFactory(application: Application, appExecutors: AppExecutors): SplashViewModelFactory {
        return SplashViewModelFactory(application, appExecutors)
    }
}